core/src/test/java/org/elasticsearch/common/settings/loader/YamlSettingsLoaderTests.java

```diff
@@ -48,42 +53,39 @@ public void testSimpleYamlSettings() throws Exception {
     }
 
-    public void testIndentationWithExplicitDocumentStart() {
-        final String yaml = "/org/elasticsearch/common/settings/loader/indentation-with-explicit-document-start-settings.yml";
-        final SettingsException e =
-                expectThrows(
-                        SettingsException.class,
-                        () -> settingsBuilder().loadFromStream(yaml, getClass().getResourceAsStream(yaml)).build());
-        assertThat(e.getMessage(), containsString("Failed to load settings"));
+    public void testIndentationWithExplicitDocumentStart() throws Exception {
+        String yaml = "/org/elasticsearch/common/settings/loader/indentation-with-explicit-document-start-settings.yml";
+        ElasticsearchParseException e = expectThrows(ElasticsearchParseException.class, () -> {
+            settingsBuilder().loadFromStream(yaml, getClass().getResourceAsStream(yaml));
+        });
+        assertTrue(e.getMessage(), e.getMessage().contains("malformed"));
     }
 
     public void testDuplicateKeysThrowsException() {
-        final String yaml = "foo: bar\nfoo: baz";
-        final SettingsException e = expectThrows(SettingsException.class, () -> settingsBuilder().loadFromSource(yaml).build());
+        String yaml = "foo: bar\nfoo: baz";
+        SettingsException e = expectThrows(SettingsException.class, () -> {
+                settingsBuilder().loadFromSource(yaml);
+        });
         assertEquals(e.getCause().getClass(), ElasticsearchParseException.class);
-        assertThat(
-                e.toString(),
-                containsString("duplicate settings key [foo] " +
-                        "found at line number [2], " +
-                        "column number [6], " +
-                        "previous value [bar], " +
-                        "current value [baz]"));
+        String msg = e.getCause().getMessage();
+        assertTrue(msg, msg.contains("duplicate settings key [foo] found"));
+        assertTrue(msg, msg.contains("previous value [bar], current value [baz]"));
     }
 
-    public void testNullValuedSettingThrowsException() {
-        final String yaml = "foo:";
-        final ElasticsearchParseException e =
-                expectThrows(ElasticsearchParseException.class, () -> new YamlSettingsLoader(false).load(yaml));
-        assertThat(e.toString(), containsString("null-valued setting found for key [foo] found at line number [1], column number [5]"));
+    public void testMissingValue() throws Exception {
+        Path tmp = createTempFile("test", ".yaml");
+        Files.write(tmp, Collections.singletonList("foo: # missing value\n"), StandardCharsets.UTF_8);
+        ElasticsearchParseException e = expectThrows(ElasticsearchParseException.class, () -> {
+            settingsBuilder().loadFromPath(tmp);
+        });
+        assertTrue(e.getMessage(), e.getMessage().contains("null-valued setting found for key [foo]"));
```

Comment:
Can we keep the line numbers in the test assertions? I think they are important to maintain.